### DISCUSSION TITLE
New Release v4.10.1 - Added the ability to save a subset of LogEntryEvent__e records, based on entry logging level

### DISCUSSION THREAD

**Discussion by jongpie** [jongpie/NebulaLogger#480] [Announcements]
# Core Unlocked Package Changes

## New `LoggerSettings__c` Field `DefaultPlatformEventStorageLoggingLevel__c`

- Reorganized all fields in the `loggerSettings` lwc to be grouped into 2 main sections: Logger Engine Settings and Log Management Settings (shown in the screenshot below)
- Resolved #475 by adding a new settings field, `LoggerSettings__c.DefaultPlatformEventStorageLoggingLevel__c` - this field can be used to optionally control which `LogEntryEvent__e` records are saved in custom objects, based on their logging level. For example, in the screenshot below

   - The existing field `LoggingLevel__c` (red box) is configured with the value `FINEST`, so all `LogEntryEvent__e` records will be published through the `EventBus`
    - The new field `DefaultPlatformEventStorageLoggingLevel__c` (green box)  is configured with the value `INFO`, so only entries with a logging level of `ERROR`, `WARN`, or `INFO` will be stored in the custom objects `Log__c` and `LogEntry__c`.

        ![image](https://user-images.githubusercontent.com/1267157/230919186-8f9858c3-368f-449d-a8b4-b186525c423b.png)

##  Bugfixes

- Fixed #478 by replacing an unnecessary query with a mock `Network` record - thanks to @rafahg for reporting this issue!

# Installation Info

## Core Unlocked Package - no namespace

Full Changelog: https://github.com/jongpie/NebulaLogger/compare/v4.10.0...v4.10.1
- SFDX CLI: `sfdx package install --wait 20 --security-type AdminsOnly --package 04t5Y0000015nYaQAI`
- Sandbox: https://test.salesforce.com/packaging/installPackage.apexp?p0=04t5Y0000015nYaQAI
- Production: https://login.salesforce.com/packaging/installPackage.apexp?p0=04t5Y0000015nYaQAI

<hr /><em>This discussion was created from the release <a href='https://github.com/jongpie/NebulaLogger/releases/tag/v4.10.1'>Added the ability to save a subset of LogEntryEvent__e records, based on entry logging level</a>.</em>

**Comment by chazwatkins** [comment]
@jongpie Does this mean I can view FINEST in the streaming console, and only the info up will be stored?

**Reply by jongpie** [reply]
@chazwatkins that's correct! There have been a few people/orgs that are worried about storage usage of the custom objects, but still need to publish more granular platform events. The new settings field can help to alleviate some of those issues. Let me know if you have any other questions!
